packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -38,6 +39,74 @@ persistCache({
     storage: window.localStorage
 });
 
+const GIFT_OPTIONS_QUERY = gql`
+    query GiftOptions {
+        gift_options {
+            include_gift_receipt
+            include_printed_card
+            gift_message
+        }
+    }
+`;
+
+const resolvers = {
+    Query: {
+        gift_options: (_, { cart_id }, { cache }) => {
```

Comment:
There is an [alternative approach](https://www.apollographql.com/docs/react/development-testing/client-schema-mocking/#gatsby-focus-wrapper) which is to define a `typeDef` that extends some existing schema, like `Cart` and then uses resolvers off that type to handle the new fields. I wonder which is "better"?

```js
const typeDefs = gql`
    extend type Cart {
        include_gift_receipt: Boolean
        include_printed_card: Boolean
        gift_message: String
    }
`;

const resolvers = {
    Cart: {
        include_gift_receipt: (...) => {...}
      ...
    }
}
```

When you use a new typedef you do several things including:

1) Inform other developers of your intent behind the new fields by defining their types ie String, Boolean, etc.
2) Lets Apollo Dev Tools (and maybe GraphIQL?) "see" the new fields so you can actually interact with them through the browser: 

[![Image from Gyazo](https://i.gyazo.com/3a152fc50e1ea5b61fff5841dd5a0b1d.png)](https://gyazo.com/3a152fc50e1ea5b61fff5841dd5a0b1d)

That said, I attempted to use `typeDefs` in the client definition and I ran into several issues with the cache initializing/persisting properly.

Reply:
I did try the typedefs and ran into all sorts of troubles. According to some GitHub issues on Apolo's page, typedefs and other UI features are still not matured enough to use. Hence I stayed away from them.